src/views/TechnicalAssistance/RequestForm/SubjectAreas.tsx

```diff
@@ -1,29 +1,563 @@
+                          <Label
+                            htmlFor="subjectAreaTechnicalReferenceArchitectureOther"
+                            hint={<div>{t('subject.hint.other')}</div>}
+                            error={!!error}
+                            className="text-normal"
+                          >
+                            {t('subject.labels.other')}
+                          </Label>
+                          {error && (
+                            <ErrorMessage>{t('errors.fillBlank')}</ErrorMessage>
+                          )}
+                          <TextInput
+                            {...field}
+                            ref={null}
+                            id="subjectAreaTechnicalReferenceArchitectureOther"
+                            type="text"
+                            validationStatus={error && 'error'}
+                          />
+                        </FormGroup>
+                      )}
+                    />
+                  )}
+              </FormGroup>
+            )}
+          />
+          */}
+
+          {fields.map(({ name, otherText, options }) => {
+            const otherOption = options[options.length - 1];
+            return (
+              <Controller
+                key={name}
+                name={name}
+                control={control}
+                render={({ field, fieldState: { error } }) => {
+                  return (
+                    <FormGroup
+                      // Use the same FormGroup error indicator for the related nested "other" field
+                      error={!!error || `${name}Other` in errors}
+                    >
+                      <Label
+                        htmlFor={name}
+                        hint={<div>{t(`subject.hint.${name}`)}</div>}
+                        error={!!error}
+                      >
+                        {t(`subject.labels.${name}`)}
+                      </Label>
+                      {error && (
+                        <ErrorMessage>{t('errors.makeSelection')}</ErrorMessage>
+                      )}
+                      <MultiSelect
+                        inputId={name}
+                        name={field.name}
+                        options={options.map(value => ({
+                          label: t(`subject.options.${name}.${value}`),
+                          value
+                        }))}
+                        initialValues={field.value}
+                        onChange={values => {
+                          field.onChange(values);
+                          // Clear errors on toggled off empty "other" fields
+                          if (!values.includes(otherOption)) {
+                            clearErrors(otherText);
+                          }
+                        }}
+                        selectedLabel={t('subject.labels.selectedTopics')}
+                      />
+                      {field.value.includes(otherOption) && (
```

Comment:
There's a Typescript error here - `field.value` is of type `TRBTechnicalReferenceArchitectureOption[] | TRBNetworkAndSecurityOption[] | (arrays of other enums)`; because there's no way to tell _which_ of those enum arrays `value` actually is, there's no valid type that `value.includes()` can accept.

I took a look at ways to fix this; I couldn't figure out any simple fix. Part of the issue is that `field` gets its type from the type parameter to `useForm`, which then gets transformed by the type code within react-hook-form. The fact that we're working with TS enums here is also somewhat limiting.

The code works despite TS reporting an error, so we're going to stick a `ts-ignore` on it and move on.